jupyterhub/templates/_helpers-netpol.tpl

```diff
@@ -0,0 +1,73 @@
+{{- define "jupyterhub.networkPolicy.standardEgressConfiguration" -}}
+{{- include "jupyterhub.networkPolicy.standardEgressConfiguration.withLeadingNewLine" . | trimPrefix "\n" }}
+{{- end }}
+
+{{- define "jupyterhub.networkPolicy.standardEgressConfiguration.withLeadingNewLine" -}}
+{{- $root := index . 0 }}
+{{- $netpol := index . 1 }}
+{{- if and ($netpol.egressAllowRules.dnsPortsPrivateIPs) (not $netpol.egressAllowRules.privateIPs) }}
+# Allow outbound connections to the DNS port in the private IP ranges
+- ports:
+    - protocol: UDP
+      port: 53
+    - protocol: TCP
+      port: 53
+  to:
+    - ipBlock:
+        cidr: 10.0.0.0/8
+    - ipBlock:
+        cidr: 172.16.0.0/12
+    - ipBlock:
+        cidr: 192.168.0.0/16
+{{- end }}
+
+{{- if $netpol.egressAllowRules.nonPrivateIPs }}
+# Allow outbound connections to non-private IP ranges
+{{- if $netpol.egressAllowRules.privateIPs }}
+# Allow outbound connections to private IP ranges (and their DNS ports)
+{{- end }}
+{{- if $netpol.egressAllowRules.cloudMetadataServer }}
+# Allow outbound connections to cloud metadata server
+{{- end }}
+- to:
+    - ipBlock:
+        cidr: 0.0.0.0/0
+        {{- if or (not $netpol.egressAllowRules.privateIPs) (not $netpol.egressAllowRules.cloudMetadataServer) }}
+        except:
+          {{- if not $netpol.egressAllowRules.privateIPs }}
+          # Don't allow outbound connections to private IP ranges
+          - 10.0.0.0/8
+          - 172.16.0.0/12
+          - 192.168.0.0/16
+          {{- end }}
+
+          {{- if not $netpol.egressAllowRules.cloudMetadataServer }}
+          # Don't allow access to the cloud metadata server
+          - {{ $root.Values.singleuser.cloudMetadata.ip }}/32
+          {{- end }}
+        {{- end }}
+{{- end }}
+
+{{- if and ($netpol.egressAllowRules.privateIPs) (not $netpol.egressAllowRules.nonPrivateIPs) }}
+# Allow outbound connections to private IP ranges (and their DNS ports)
```

Comment:
Mismatched comment?

Reply:
The "(and their DNS ports)" was a hint about `dnsPortsPrivateIPs` becomes irrelevant, as it automatically accepts communication to the DNS ports by allowing connections to these IPs independent of ports.

Do you have a change suggestion? (preferably sticking to a single line)

Reply:
Just remove the comment about 'and their DNS ports'?